dhctl/pkg/operations/destroy/destroy.go

```diff
@@ -505,7 +505,7 @@ func (d *StaticMastersDestroyer) switchToNodeuser(settings *session.Session) err
 
 	log.DebugLn("Private key written")
 
-	if !app.SSHLegacyMode {
+	if app.SSHModernMode || len(app.SSHPrivateKeys) == 0 {
```

Comment:
Please move this condition to function in `sshclient/init.go`

For example with name `NeedStopModernClient`